test/boost/encryption_at_rest_test.cc

```diff
@@ -1077,41 +1077,49 @@ static future<> test_broken_encrypted_commitlog(const test_provider_args& args,
     fake_proxy proxy(host);
+    std::exception_ptr p;
+    try {
+        auto [scopts, yaml] = make_opts(proxy);
 
-    auto [scopts, yaml] = make_opts(proxy);
-
-    test_provider_args args{
-        .tmp = tmp,
-        .extra_yaml = yaml,
-        .n_tables = 10, 
-        .before_create_table = [&](auto& env) {
-            // turn off proxy. all key resolution after this should fail
-            proxy.enable(false);
-            // wait for key cache expiry.
-            seastar::sleep(10ms).get();
-            // ensure commitlog will create a new segment on write -> eventual write failure
-            env.db().invoke_on_all([](replica::database& db) {
-                return db.commitlog()->force_new_active_segment();
-            }).get();
-        },
-        .on_insert_exception = [&](auto&&) {
-            // once we get the exception we have to enable key resolution again, 
-            // otherwise we can't shut down cql test env.
-            proxy.enable(true);
-        },
-        .timeout = timeout_config{
-            // set really low write timeouts so we get a failure (timeout)
-            // when we fail to write to commitlog
-            100ms, 100ms, 100ms, 100ms, 100ms, 100ms, 100ms
-        },
-    };
+        test_provider_args args{
+            .tmp = tmp,
+            .extra_yaml = yaml,
+            .n_tables = 10, 
+            .before_create_table = [&](auto& env) {
+                // turn off proxy. all key resolution after this should fail
+                proxy.enable(false);
+                // wait for key cache expiry.
+                seastar::sleep(10ms).get();
+                // ensure commitlog will create a new segment on write -> eventual write failure
+                env.db().invoke_on_all([](replica::database& db) {
+                    return db.commitlog()->force_new_active_segment();
+                }).get();
+            },
+            .on_insert_exception = [&](auto&&) {
+                // once we get the exception we have to enable key resolution again, 
+                // otherwise we can't shut down cql test env.
+                proxy.enable(true);
+            },
+            .timeout = timeout_config{
+                // set really low write timeouts so we get a failure (timeout)
+                // when we fail to write to commitlog
+                100ms, 100ms, 100ms, 100ms, 100ms, 100ms, 100ms
+            },
+        };
 
-    BOOST_REQUIRE_THROW(
-        co_await test_broken_encrypted_commitlog(args, scopts);
-        , exceptions::mutation_write_timeout_exception
-    );
+        BOOST_REQUIRE_THROW(
+            co_await test_broken_encrypted_commitlog(args, scopts);
+            , exceptions::mutation_write_timeout_exception
+        );
+    } catch (...) {
+        p = std::current_exception();
+    }
 
     co_await proxy.stop();
+
+    if (p) {
+        std::rethrow_exception(p);
```

Comment:
Why not just fail the test?

Reply:
Throwing the exception will fail the test. And propagate the info. Automatically. Separation of concern.